web/src/compose_tooltips.ts

```diff
@@ -14,10 +14,55 @@ import {pick_empty_narrow_banner} from "./narrow_banner.ts";
 import * as narrow_state from "./narrow_state.ts";
 import * as popover_menus from "./popover_menus.ts";
 import {realm} from "./state_data.ts";
-import {EXTRA_LONG_HOVER_DELAY, INSTANT_HOVER_DELAY, LONG_HOVER_DELAY} from "./tippyjs.ts";
+import {
+    EXTRA_LONG_HOVER_DELAY,
+    INSTANT_HOVER_DELAY,
+    LONG_HOVER_DELAY,
+    get_tooltip_content,
+} from "./tippyjs.ts";
 import {parse_html} from "./ui_util.ts";
 import {user_settings} from "./user_settings.ts";
 
+let compose_tooltips: tippy.Instance[] | null = null;
+let compose_tooltip_singleton: tippy.CreateSingletonInstance | null = null;
+
+export function initialize_compose_tooltips(): void {
+    // Clean up existing instances first
+    if (compose_tooltip_singleton) {
+        compose_tooltip_singleton.destroy();
+        if (compose_tooltips) {
+            for (const tippy_instance of compose_tooltips) {
+                if (!tippy_instance.state.isDestroyed) {
+                    tippy_instance.destroy();
+                }
+            }
+        }
+    }
+
+    compose_tooltips = tippy.default(".compose_button_tooltip", {
+        trigger: "mouseenter",
+        placement: "top",
+    });
+
+    compose_tooltip_singleton = tippy.createSingleton(compose_tooltips, {
+        delay: LONG_HOVER_DELAY,
+        onTrigger(instance, event) {
+            const currentTarget = event.currentTarget;
+            if (currentTarget instanceof HTMLElement) {
+                const content = get_tooltip_content(currentTarget);
+                if (content) {
+                    instance.setContent(content);
+                }
+                if (currentTarget.classList?.contains("disabled-on-hover")) {
+                    instance.setProps({delay: [100, 250]});
+                } else {
+                    instance.setProps({delay: [750, 250]});
+                }
+            }
+        },
+    });
+}
```

Comment:
Let's have this block of code be a separate commit, since this is responsible for the actual singleton implementation.

Reply:
Let's put the rest of the code in a prep commit prior to this one.

Reply:
In the singleton commit, let's define these delay values as constants in `web/src/tippyjs.ts` with appropriate naming such as `SINGLETON_INSTANT_HOVER_DELAY` and `SINGLETON_LONG_HOVER_DELAY` with supporting comments describing why we need these extra delays for the singletons.